modules/dxp/apps/commerce/commerce-shop-by-diagram/commerce-shop-by-diagram-web/src/main/resources/META-INF/resources/js/AdminTooltip.js

```diff
@@ -15,36 +15,67 @@ import ClayCard from '@clayui/card';
 import ClayDropDown from '@clayui/drop-down';
 import ClayForm, {ClayInput, ClayRadio, ClayRadioGroup} from '@clayui/form';
 import PropTypes from 'prop-types';
-import React, {useEffect, useState} from 'react';
+import React, {useEffect, useRef, useState} from 'react';
+
+import {searchDiagrams, searchSkus} from './utilities/utilities';
 
 const AdminTooltip = ({
 	deletePin,
 	namespace,
-	searchSkus,
 	setRemovePinHandler,
 	setShowTooltip,
 	showTooltip,
-	skus,
 	updatePin,
 }) => {
+	const [active, setActive] = useState(false);
+	const [products, setProducts] = useState(null);
 	const [pinPositionLabel, setPinPositionLabel] = useState(
 		showTooltip.details.label
 	);
-	const [linkedValue, setLinkedValue] = useState(
-		showTooltip.details.linked_to_sku
-	);
-	const [sku, setSku] = useState(showTooltip.details.sku);
+	const node = useRef();
+	const dropdownNode = useRef();
+	const [query, setQuery] = useState('');
+	const [linkedValue, setLinkedValue] = useState('sku');
```

Comment:
To avoid refactor regressions, it is probably best to store this `'sku'` value as a variable at the top of the file.

```js
const SKU = 'sku';
```